### PR TITLE
Module 7 redux

### PR DESCRIPTION
**Pull request by SergBlack**

### CODE REVIEW

File: src/components/Main/index.jsx [hunk 1]
```diff
@@ -1,17 +1,34 @@
...
+  selectTotalAmount,
+  selectCurrentSort,
+  selectSortParams,
+  selectCurrentFilter,
+  selectFilterParams,
+} from '@/redux/selectors';
+import {
+  loadMovies,
+  setCurrentFilter,
+  setCurrentSort, setFilterParams,
```
Comment (mmarkelov) [inline src/components/Main/index.jsx]:
```js
setCurrentSort,
setFilterParams,
```

looks like you missed new line

File: src/components/Main/index.jsx [hunk 2]
```diff
@@ -28,44 +45,62 @@ const StyledFilterBlock = styled.div`
...
+    dispatch(setSortParams(params));
+  };
 
   const onFilterClick = (value) => {
-    setCurrentFilter(value);
+    dispatch(setCurrentFilter(value));
+    if (value === 'All') {
+      dispatch(setFilterParams({}));
+    } else {
+      dispatch(setFilterParams({ searchBy: 'genres', filter: value.toLowerCase() }));
```
Comment (mmarkelov) [inline src/components/Main/index.jsx]: not an error, but i think it'll be better to use ternary operator
```js
const filterParams = value === 'All' ? {} : { searchBy: 'genres', filter: value.toLowerCase() };
dispatch(setFilterParams(filterParams));
```

File: src/forms/DeleteMovieForm.jsx [hunk 3]
```diff
@@ -11,11 +15,18 @@ const ButtonWrapper = styled.div`
...
-const DeleteMovieForm = ({ formTitle }) => {
-  const handleSubmit = (e) => {
-    e.stopPropagation();
-    // eslint-disable-next-line no-alert
-    alert('Movie deleted');
+const DeleteMovieForm = ({ formTitle, id, close }) => {
+  const dispatch = useDispatch();
+  const history = useHistory();
+
+  const afterSuccess = () => {
```
Comment (mmarkelov) [inline src/forms/DeleteMovieForm.jsx]: I'm not sure that `afterSuccess` is good name for this function. Maybe it's better to use `reset`?

File: src/forms/DeleteMovieForm.jsx [hunk 4]
```diff
@@ -11,11 +15,18 @@ const ButtonWrapper = styled.div`
...
+    close();
+    history.push('/');
+    dispatch(loadMovies());
+  };
+
+  const handleSubmit = () => {
+    dispatch(deleteMovie(id, () => afterSuccess()));
```
Comment (mmarkelov) [inline src/forms/DeleteMovieForm.jsx]: not sure about it. But maybe you can use it like this
```js
dispatch(deleteMovie(id, afterSuccess));
```

File: src/forms/MovieForm.jsx [hunk 5]
```diff
@@ -1,92 +1,85 @@
...
 import styled from 'styled-components';
 
+import { selectEditedMovie } from '@/redux/selectors';
+import {
+  setEditedMovieToForm,
+  resetForm,
+  handleFormInput,
+  handleFormSelect,
+  addMovie,
+  updateMovie, loadMovies,
```
Comment (mmarkelov) [inline src/forms/MovieForm.jsx]: missed new line

File: src/forms/MovieForm.jsx [hunk 6]
```diff
@@ -1,92 +1,85 @@
...
-  };
+  }, [dispatch, movie]);
 
   const handleReset = useCallback(
-    () => { dispatch({ type: 'RESET_FORM' }); },
+    () => { dispatch(resetForm()); },
     [dispatch],
   );
 
+  const afterSuccess = (id) => {
```
Comment (mmarkelov) [inline src/forms/MovieForm.jsx]: also here. It's better to rename this function

File: src/helpers/objectParamsToQueryString.js [hunk 7]
```diff
@@ -0,0 +1,6 @@
+const objectParamsToQueryString = (params) => {
+  const queryString = Object.keys(params).map((key) => `${key}=${params[key]}`).join('&');
```
Comment (mmarkelov) [inline src/helpers/objectParamsToQueryString.js]: Maybe it's better to use `Object.entries`?
```js
const queryString = Object.entries(params).map((key, value) => `${key}=${value}`).join('&');
```

File: src/redux/actions.js [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+  SET_CURRENT_SORT,
+  SET_FILTER_PARAMS,
+  SET_CURRENT_FILTER,
+  SET_EDITED_MOVIE_TO_FORM,
+  RESET_FORM,
+  HANDLE_INPUT,
+  HANDLE_SELECT,
+} from '@/redux/types';
+
+const API_URL = 'http://127.0.0.1:4000/';
```
Comment (mmarkelov) [inline src/redux/actions.js]: Also not an error. But maybe it's cleaner to use '/' for your path?
```js
const API_URL = 'http://127.0.0.1:4000';
...
const response = await axios.get(`${API_URL}/movies${params}`, { headers });
```

File: src/redux/reducers/movieFormReducer.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        return { ...state, genres: filteredGenres };
+      }
+      return { ...state, genres: [...state.genres, newGenre] };
+    }
+
+    case SET_EDITED_MOVIE_TO_FORM: {
+      return { ...action.payload };
+    }
+
+    default: return state;
```
Comment (mmarkelov) [inline src/redux/reducers/movieFormReducer.js]:
```js
default: 
  return state;
```

File: src/components/Main/index.jsx (hunk 1 above)
Comment (SergBlack) [reply src/components/Main/index.jsx]: Thanks. Fixed)

File: src/components/Main/index.jsx (hunk 2 above)
Comment (SergBlack) [reply src/components/Main/index.jsx]: Refactor. Thanks)

File: src/forms/DeleteMovieForm.jsx (hunk 3 above)
Comment (SergBlack) [reply src/forms/DeleteMovieForm.jsx]: Rename to resetForm) Thank you!

File: src/forms/DeleteMovieForm.jsx (hunk 4 above)
Comment (SergBlack) [reply src/forms/DeleteMovieForm.jsx]: Fixed!) Thanks!

File: src/forms/MovieForm.jsx (hunk 5 above)
Comment (SergBlack) [reply src/forms/MovieForm.jsx]: Fixed!) Thanks!

File: src/forms/MovieForm.jsx (hunk 6 above)
Comment (SergBlack) [reply src/forms/MovieForm.jsx]: Fixed!) Thanks!

File: src/helpers/objectParamsToQueryString.js (hunk 7 above)
Comment (SergBlack) [reply src/helpers/objectParamsToQueryString.js]: Refactor! Thanks)

File: src/redux/reducers/movieFormReducer.js (hunk 9 above)
Comment (SergBlack) [reply src/redux/reducers/movieFormReducer.js]: Fixed) Thanks!

File: src/redux/actions.js (hunk 8 above)
Comment (SergBlack) [reply src/redux/actions.js]: Fixed) Thanks!

**Pull request merged**